File: SyntaxGenDotNet.VisualBasic/Operators.vb
Imports SyntaxGenDotNet.Syntax
Imports SyntaxGenDotNet.Syntax.Tokens

Friend Module Operators
    ''' <summary>
    '''     The <c>=</c> operator.
    ''' </summary>
    Public ReadOnly Property Assignment As OperatorToken
        Get
            Return New OperatorToken("=", False) With {
                .TrailingWhitespace = WhitespaceTrivia.Space
                }
        End Get
    End Property

    ''' <summary>
    '''     The <c>&gt;</c> operator.
    ''' </summary>
    Public ReadOnly Property CloseChevron As OperatorToken
        Get
            Return New OperatorToken(">")

            ''' <summary>
        End Get
    End Property

    '''     The <c>)</c> operator.
    ''' </summary>
    Public ReadOnly Property CloseParenthesis As OperatorToken
        Get
            Return New OperatorToken(")")

            ''' <summary>
        End Get
    End Property

    '''     The <c>:</c> operator.
    ''' </summary>
    Public ReadOnly Property Colon As OperatorToken
        Get
            Return New OperatorToken(":") With {
                .LeadingWhitespace = WhitespaceTrivia.Space
                }
        End Get
    End Property

    ''' <summary>
    '''     The <c>,</c> operator.
    ''' </summary>
    Public ReadOnly Property Comma As OperatorToken
        Get
            Return New OperatorToken(",") With {
                .TrailingWhitespace = WhitespaceTrivia.Space
                }
        End Get
    End Property

    ''' <summary>
    '''     The <c>.</c> operator.
    ''' </summary>
    Public ReadOnly Property Dot As OperatorToken
        Get
            Return New OperatorToken(".")
        End Get
    End Property

    ''' <summary>
    '''     The <c>&lt;</c> operator.
    ''' </summary>
    Public ReadOnly Property OpenChevron As OperatorToken
        Get
            Return New OperatorToken("<")
        End Get
    End Property

    ''' <summary>
    '''     The <c>(</c> operator.
    ''' </summary>
    Public ReadOnly Property OpenParenthesis As OperatorToken
        Get
            Return New OperatorToken("(")
        End Get
    End Property
End Module

File: SyntaxGenDotNet.CSharp/Operators.cs
using SyntaxGenDotNet.Syntax;
using SyntaxGenDotNet.Syntax.Tokens;

namespace SyntaxGenDotNet.CSharp;

internal static class Operators
{
    /// <summary>
    ///     The <c>=</c> operator.
    /// </summary>
    public static readonly OperatorToken Assignment = new("=", false)
    {
        LeadingWhitespace = WhitespaceTrivia.Space, TrailingWhitespace = WhitespaceTrivia.Space
    };

    /// <summary>
    ///     The <c>=</c> operator.
    /// </summary>
    public static readonly OperatorToken Asterisk = new("*") {TrailingWhitespace = WhitespaceTrivia.Space};

    /// <summary>
    ///     The <c>]</c> operator.
    /// </summary>
    public static readonly OperatorToken CloseBracket = new("]");

    /// <summary>
    ///     The <c>&gt;</c> operator.
    /// </summary>
    public static readonly OperatorToken CloseChevron = new(">");

    /// <summary>
    ///     The <c>)</c> operator.
    /// </summary>
    public static readonly OperatorToken CloseParenthesis = new(")");

    /// <summary>
    ///     The <c>:</c> operator.
    /// </summary>
    public static readonly OperatorToken Colon = new(":") {Whitespace = WhitespaceTrivia.Space};

    /// <summary>
    ///     The <c>,</c> operator.
    /// </summary>
    public static readonly OperatorToken Comma = new(",") {TrailingWhitespace = WhitespaceTrivia.Space};

    /// <summary>
    ///     The <c>.</c> operator.
    /// </summary>
    public static readonly OperatorToken Dot = new(".");

    /// <summary>
    ///     The <c>[</c> operator.
    /// </summary>
    public static readonly OperatorToken OpenBracket = new("[");

    /// <summary>
    ///     The <c>&lt;</c> operator.
    /// </summary>
    public static readonly OperatorToken OpenChevron = new("<");

    /// <summary>
    ///     The <c>(</c> operator.
    /// </summary>
    public static readonly OperatorToken OpenParenthesis = new("(");

    /// <summary>
    ///     The <c>?</c> operator.
    /// </summary>
    public static readonly OperatorToken QuestionMark = new("?") {TrailingWhitespace = WhitespaceTrivia.Space};

    /// <summary>
    ///     The <c>;</c> operator.
    /// </summary>
    public static readonly OperatorToken Semicolon = new(";");
}
